annotations/tei_utils.py

```diff
@@ -0,0 +1,585 @@
+"""
+TEI-XML Integration utilities for Vogon.
+
+1. Detects content type (TEI, XML, plain text)
+2. Parses TEI-XML
+3. Converts TEI to a more "clean" HTML representation:
+   - Minimizes raw TEI markup
+   - Preserves structural layout (paragraphs, line/page breaks, headings, etc.)
+   - Provides a simpler display of editorial tags (e.g., <choice>, <orig>, <reg>)
+4. Maintains xpath references for annotation
+"""
+
+from lxml import etree
+from django.utils.safestring import mark_safe
+import re
+
+# ===== Content Type Detection =====
+
+def detect_content_type(text_content):
+    """
+    Detect the content type of a text by examining its structure.
+
+    Returns a dictionary with:
+        - is_xml (bool)
+        - is_tei (bool)
+        - content_type (str) = 'text/xml+tei', 'text/xml', or 'text/plain'
+    """
+    is_xml = False
+    is_tei = False
+
+    if text_content.strip().startswith('<?xml') or text_content.strip().startswith('<'):
```

Comment:
The first condition is included in the second, isn't it? If the text starts with "<?xml", then it also starts with "<".